src/qt/walletcontroller.cpp

```diff
@@ -366,3 +366,40 @@ void LoadWalletsActivity::load()
         QTimer::singleShot(0, this, [this] { Q_EMIT finished(); });
     });
 }
+
+RestoreWalletActivity::RestoreWalletActivity(WalletController* wallet_controller, QWidget* parent_widget)
+    : WalletControllerActivity(wallet_controller, parent_widget)
+{
+}
+
+void RestoreWalletActivity::restore(const std::string& backup_file, const std::string& wallet_name)
+{
+    QString name = QString::fromStdString(wallet_name);
+
+    showProgressDialog(
+        tr("Restore Wallet"),
+        tr("Restoring Wallet <b>%1</b>…").arg(name.toHtmlEscaped())
+    );
+
+    QTimer::singleShot(0, worker(), [this, backup_file, wallet_name] {
+       std::unique_ptr<interfaces::Wallet> wallet = node().walletClient().restoreWallet(backup_file, wallet_name, m_error_message, m_warning_message);
+
+        if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));
+
+        QTimer::singleShot(0, this, &RestoreWalletActivity::finish);
```

Comment:
Standardise spacing:

```suggestion
        std::unique_ptr<interfaces::Wallet> wallet = node().walletClient().restoreWallet(backup_file, wallet_name, m_error_message, m_warning_message);

        if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));

        QTimer::singleShot(0, this, &RestoreWalletActivity::finish);
```
<p>

**Suggestion:** How about using the auto keyword here? That would make the code simpler and more readable.
@hebasto, what do you think about this suggestion?

```suggestion
       auto wallet = node().walletClient().restoreWallet(backup_file, wallet_name, m_error_message, m_warning_message);

        if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));

        QTimer::singleShot(0, this, &RestoreWalletActivity::finish);
```